docs/refactoring/course-read-model/scripts/amplitude_extract.py

```
@@ -0,0 +1,120 @@
+
+    for s, e in chunks:
+        dest = os.path.join(out_dir, f"w{s.strftime('%Y%m%d')}.zip")
+        if os.path.exists(dest) and os.path.getsize(dest) > 0:
+            print(f"{dest}: skip (exists)", flush=True)
+            continue
+        url = (f"https://amplitude.com/api/2/export"
+               f"?start={s.strftime('%Y%m%d')}T00&end={e.strftime('%Y%m%d')}T23")
+        for attempt in range(3):
+            req = urllib.request.Request(url, headers={"Authorization": f"Basic {auth}"})
+            try:
+                with urllib.request.urlopen(req, timeout=300) as resp:
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)